pkg/datapath/linux/node.go

```diff
@@ -1446,6 +1470,101 @@ func (n *linuxNodeHandler) NodeNeighborRefresh(ctx context.Context, nodeToRefres
 	}
 }
 
+// NodeCleanNeighbors cleans all neighbor entries for previously used network
+// link interfaces. It should be used when the agent changes the state from
+// `n.enableNeighDiscovery = true` to `n.enableNeighDiscovery = false`.
+func (n *linuxNodeHandler) NodeCleanNeighbors() {
+	devToClean, err := loadNeighLinks(option.Config.StateDir)
+	if err != nil {
+		log.WithError(err).Error("Unable to load old neigh links configuration." +
+			" Previous running state needs to be cleaned up manually.")
+		return
+	}
+	if len(devToClean) == 0 {
+		return
+	}
+	// Delete the file after cleaning up neighbor list.
+	defer os.Remove(filepath.Join(option.Config.StateDir, neighListFileName))
+
+	n.mutex.Lock()
+	defer n.mutex.Unlock()
+
+	for _, linkName := range devToClean {
+		l, err := netlink.LinkByName(linkName)
+		if err != nil {
+			if _, ok := err.(netlink.LinkNotFoundError); !ok {
+				log.WithError(err).WithFields(logrus.Fields{
+					logfields.Device: linkName,
+				}).Error("Unable to GC old neighbor entries for network device")
```

Comment:
s/Unable to GC old neighbor entries.../Unable to remove PERM ARP entries of network device/